tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -207,3 +208,98 @@ def test_infinite_retention(self):
         producer.wait()
         rand_consumer.shutdown()
         rand_consumer.wait()
+
+
+class ShadowIndexingWhileBusyTest(PreallocNodesTest):
+    # With SI enabled, run common operations against a cluster
+    # while the system is under load (busy).
+    # This class ports Test6 from https://github.com/redpanda-data/redpanda/issues/3572
+    # into ducktape.
+    segment_size = 2**30
+    topics = [TopicSpec(partition_count=5)]
+
+    def __init__(self, test_context):
+        self._si_settings = SISettings(
+            log_segment_size=self.segment_size,
+            cloud_storage_cache_size=20 * 2**30,
+            cloud_storage_reconciliation_interval_ms=500,
+            cloud_storage_max_connections=5,
+            cloud_storage_enable_remote_read=False,
+            cloud_storage_enable_remote_write=False)
+
+        super(ShadowIndexingWhileBusyTest,
+              self).__init__(test_context=test_context,
+                             node_prealloc_count=1,
+                             num_brokers=3,
+                             si_settings=self._si_settings)
+
+    @ok_to_fail  # broken pipe inconsistently causes failure from cloud_storage/remote.cc:108
+    @cluster(num_nodes=4, log_allow_list=KGO_ALLOW_LOGS)
+    def test_create_or_delete_topics_while_busy(self):
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        await_bucket_creation(self.redpanda,
+                              self._si_settings.cloud_storage_bucket)
+
+        # Remote write/read and retention set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+
+        # 100k messages of size 2**18
+        # is ~24GB of data.
+        msg_size = 2**18
+        msg_count = 100000
+        timeout = 600
```

Comment:
This timeout is assuming approx 40MB/s throughput, which is a bit optimistic (!) for our oversubscribed test nodes.  For tests like ManyPartitionsTest, I found a more pessimistic 10MB/s worked well.